Mathlib/Algebra/Category/Grp/Injective.lean

```diff
@@ -54,6 +56,31 @@ theorem Module.Baer.of_divisible [DivisibleBy A ℤ] : Module.Baer ℤ A := fun
   rw [map_zsmul, LinearMap.toSpanSingleton_apply, DivisibleBy.div_cancel gₘ h0, ← map_zsmul g,
     SetLike.mk_smul_mk]
 
+namespace Module
+variable (R Q : Type*) [Ring R] [AddCommGroup Q] [Module R Q]
+
+theorem injective_object_of_injective_module [inj : Injective R Q] :
+    CategoryTheory.Injective (ModuleCat.of R Q) where
+  factors g f m :=
+    have ⟨l, h⟩ := inj.out f ((ModuleCat.mono_iff_injective f).mp m) g
+    ⟨l, LinearMap.ext h⟩
+
+theorem injective_module_of_injective_object
+    [inj : CategoryTheory.Injective <| ModuleCat.of R Q] :
+    Module.Injective R Q where
+  out X Y _ _ _ _ f hf g := by
+    have : CategoryTheory.Mono (ModuleCat.asHom f) := (ModuleCat.mono_iff_injective _).mpr hf
+    obtain ⟨l, rfl⟩ := inj.factors (ModuleCat.asHom g) (ModuleCat.asHom f)
+    exact ⟨l, fun _ ↦ rfl⟩
+
+theorem injective_iff_injective_object :
+    Module.Injective R Q ↔
+    CategoryTheory.Injective (ModuleCat.of R Q) :=
+  ⟨fun _ => injective_object_of_injective_module R Q,
+   fun _ => injective_module_of_injective_object R Q⟩
```

Comment:
Maybe we should create another file for these results, just like [the case of projective modules](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Category/ModuleCat/Projective.html).

Note that there is an asymmetry between the projective and injective APIs, which is that `Module.Projective` is equivalent to the universe polymorphic lifting property (stronger than `CategoryTheory.Projective`), while `Module.Injective` is only the lifting property in the universe of the module (equivalent to `CategoryTheory.Injective`). For this reason, [IsProjective.iff_projective](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Category/ModuleCat/Projective.html#IsProjective.iff_projective) requires `P : Type (max u v)`, and [Module.Injective.extension_property](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Module/Injective.html#Module.Injective.extension_property) has a `Small` assumption. Arguably, the real analogue of [Module.projective_lifting_property](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Module/Projective.html#Module.projective_lifting_property) should be [Module.Baer.extension_property](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Module/Injective.html#Module.Baer.extension_property), where `Module.Baer` is equivalent to the universe polymorphic extension property. For this reason, instead of providing instances of `Module.Injective`, we should probably make `Module.Baer` a class and provide instances of it.

cc @jjaassoonn

Reply:
> Maybe we should create another file for these results, just like [the case of projective modules](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Category/ModuleCat/Projective.html).

`Mathlib/Algebra/Category/ModuleCat/Injective` already exists, and is part of this PR. It contains a comment at the top of the file explaining why these results cannot live there.

Reply:
Thanks for pointing out that the file exists; however the file should probably be renamed `EnoughInjectives` and we could make the `Injective` file contain only these three results (and hopefully, import only Algebra/Module/Injective.lean and CategoryTheory/Preadditive/Injective.lean). The docstring talks about enough injectives and I don't think it has anything to do with these three results.

Reply:
Done